### ISSUE TITLE
Alt-F2 window/popup stuck when entering non-existent command

### ISSUE THREAD

**Issue by bjd-pfq** [linuxmint/cinnamon#8593]
```
 * Cinnamon version (cinnamon --version)
Cinnamon 3.6.7

 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...)
Linux Mint 18.3 Sylvia

 * Graphics hardware *and* driver used
01:00.0 VGA compatible controller: NVIDIA Corporation GP104 [GeForce GTX 1070] (rev a1)

 * 32 or 64 bit
64

 * Attach /home/<username>/.xsession-errors, or /var/log/syslog
n/a
 ```

**Issue**
I open Alt-F2 to run a command. The screen greys out and I see what I call a command-line. I enter a non-existent command, let's say 'foo', and press RETURN. A message is displayed "Failed to execute child process foo, no such file or directory", next to  an icon with a cross in it, apparently signifying "click me" to remove this error message.
No such functionality is provided though. I can click all I want, on the cross, next to the cross, next to the  'popup' with the command line in which I entered 'foo', nothing happens to remove the message and get my screen back.
I can only get my screen back by backspacing the line where I entered 'foo', clearing it that way, and press RETURN. My normal desktop re-appears, and an excess nemo-window appears, something I hadnn't asked for.

**Steps to reproduce**
See above

**Expected behaviour**
Clicking on the aforementioned cross does what it signals to promise to do. Hard to implement?

**Other information**

**Comment by SketchNI** [comment]
In the latest cinnamon (`3.8.9`), this has been changed to a red exclamation circle.

![image](https://user-images.githubusercontent.com/11874768/58707968-54194080-83ae-11e9-97b1-4191ef832d53.png)

**Comment by Odyseus** [comment]
Hello, everybody.

A couple of clarifications:

1. The dialog doesn't get stuck, it stays opened so it can display the message that you see informing you that there has been an error executing the command that you tried to execute.
2. You close the dialog the same way that you close a dialog in any GUI toolkit and in any platform, by pressing the <kbd>Escape</kbd> key.
3. The *icon with a cross in it* that you see is provided by the icon theme that you have chosen is called **dialog-error** and it doesn't signify *click me*, it signifies *there has been an error*.

So, in a nutshell, the dialog works as it's supposed to work.

**Comment by bjd-pfq** [comment]
"You close the dialog the same way that you close a dialog in any GUI toolkit and in any platform, by pressing the Escape key".

Thanks for informing me (btw., I'm a paraplegic, and with mouse in hand reaching for the keyboard is quite an exertion, and seems illogical and not really user friendly regardless). If what you say is correct, I have dozens of bugs regarding popups in other applications, but I won't bother you with those.

**Comment by smurphos** [comment]
> I can only get my screen back by backspacing the line where I entered 'foo', clearing it that way, and press RETURN. My normal desktop re-appears, and an excess nemo-window appears, something I hadnn't asked for.

I'd call this bit a bug (the opening nemo with an empty run dialog command). I can replicate in Cinnamon 4.0.10 just by opening the run dialog and pressing enter.

A close button on the dialog would probably be nice to have as escape clearly isn't intuitive (or convenient for all as bjd-pfq points out).

**Comment by JosephMcc** [comment]
This behaves differently on git master. Typing "foo" in the dialog simply causes it to close and nothing happens. With an empty dialog it simply closes also.

**Comment by JosephMcc** [comment]
I can replicate the nemo opening thing.

**Comment by ItzSwirlz** [comment]
not sure if this is still a thing but it's likely a duplicate of #9875

**Comment by JulianGro** [comment]
It's not a duplicate. The other issue talks about the run dialogue crashing. This one about it not having a close button.
I believe you can just click anywhere outside of it to close it though.
Between that, pressing escape, and just pressing enter without a command, it seems simple enough to figure out how to close it.

**Comment by JosephMcc** [comment]
The close button was discussed among the devs previously. It won't be getting added. Can no longer reproduce it opening the extra Nemo window either.
